packages/apps/spaces/app/organizations/src/components/Actions/Actions.tsx

```diff
@@ -1,58 +1,107 @@
-        leftIcon={<Archive />}
-        onClick={onArchiveOrganizations}
-        _hover={{
-          bg: 'gray.800',
-        }}
-      >
-        {`Archive ${
-          selectCount > 1 ? `these ${selectCount}` : ' this company'
-        }`}
-      </Button>
-      {selectCount > 1 && (
-        <Button
-          bg='gray.700'
-          color='white'
-          leftIcon={<Copy07 />}
-          _hover={{
-            bg: 'gray.800',
-          }}
-          onClick={() => {
-            onMergeOrganizations();
+export const TableActions = memo(
+  ({
+    table,
+    isArchiving,
+    onMergeOrganizations,
+    onArchiveOrganizations,
+  }: TableActionsProps) => {
+    const { isOpen, onOpen, onClose } = useDisclosure();
+    const [targetIndex, setTargetIndex] = useState<string | null>(null);
+
+    const selection = table.getState().rowSelection;
+    const selectCount = Object.keys(selection).length;
+
+    useEffect(() => {
+      if (selectCount === 1) {
+        const [index] = Object.entries(selection)[0];
+        setTargetIndex(index);
+      }
+      if (selectCount < 1) {
+        setTargetIndex(null);
+      }
+    }, [selectCount]);
+
+    if (!selectCount && !targetIndex) return null;
+
+    return (
+      <>
+        <Center left='50%' position='absolute' bottom='32px'>
+          <ButtonGroup size='md' isAttached left='-50%' position='relative'>
+            <Button
+              bg='gray.700'
+              color='white'
+              leftIcon={<Archive />}
+              onClick={onOpen}
+              _hover={{
+                bg: 'gray.800',
+              }}
+            >
+              {`Archive ${
+                selectCount > 1 ? `these ${selectCount}` : ' this company'
+              }`}
+            </Button>
+            {selectCount > 1 && (
+              <Button
+                bg='gray.700'
+                color='white'
+                leftIcon={<Copy07 />}
+                _hover={{
+                  bg: 'gray.800',
+                }}
+                onClick={() => {
+                  onMergeOrganizations(targetIndex as string, selection);
+                  table.resetRowSelection();
+                }}
+              >
+                {`Merge these ${selectCount}`}
+              </Button>
+            )}
+          </ButtonGroup>
+        </Center>
+
+        <ConfirmDeleteDialog
+          isOpen={isOpen}
+          icon={<Archive />}
+          onClose={onClose}
+          isLoading={isArchiving}
+          confirmButtonLabel={'Archive'}
+          onConfirm={() => {
+            onArchiveOrganizations(selection);
+            onClose();
             table.resetRowSelection();
           }}
-        >
-          {`Merge these ${selectCount}`}
-        </Button>
-      )}
-    </ButtonGroup>
-  );
-};
+          label={`Archive selected ${
+            selectCount === 1 ? 'organization' : 'organizations'
+          }?`}
+        />
+      </>
+    );
+  },
+  (prev, next) => {
```

Comment:
the condition is missing `isArchiving` check

Reply:
Actually I'll refactor this a bit to make it even better. I'll move mutation handlers inside the Actions compoment so that there's no need to pass these fns through props.

Reply:
isArchiving is a boolean attr, not function, anyway let me know when it's done